static/app/views/automations/hooks/utils.tsx

```diff
@@ -12,3 +14,25 @@ export function useAutomationActions(automation: Automation): ActionType[] {
     ),
   ] as ActionType[];
 }
+
+export function useConnectedIds() {
+  const [connectedIds, setConnectedIds] = useState<Set<string>>(() => {
+    const stored = localStorage.getItem('connectedIds');
+    return stored ? new Set(JSON.parse(stored)) : new Set();
+  });
+
+  const toggleConnected = (id: string) => {
+    setConnectedIds(prev => {
+      const newSet = new Set(prev);
+      if (newSet.has(id)) {
+        newSet.delete(id);
+      } else {
+        newSet.add(id);
+      }
+      localStorage.setItem('connectedIds', JSON.stringify(Array.from(newSet)));
```

Comment:
I'd consider either accepting a required `key` parameter to use as the `localStorage` key or leveraging the [`useId()`](https://react.dev/reference/react/useId) hook to generate one. Currently, interacting with multiple components that use this hook would overwrite earlier states.

Reply:
update: i realized we can't just generate a unique key for every use, since we want step 1 and 2 of automation creation to maintain the same list of `connectedIds`. i've updated this hook to take a `key` parameter, but made a const key for steps 1 and 2 of automation creation to share (see https://github.com/getsentry/sentry/pull/91408/commits/ee135ba79ddb9d5bbbf1fcbbfccfa3b8fed0a77a). let me know if there is a better way for 2 views to share a key!